### PR TITLE
Update documentation for Quarkus powered Keycloak/Red Hat build of Keycloak

### LINKED ISSUE #2499: Update Keycloak Authentication docs
As mentioned in [Discourse](https://docs.theforeman.org/nightly/Installing_Server/index-katello.html#Configuring_Project_with_Keycloak_Authentication_keycloak-general) it would be needed to update the Keycloak documentation, as since a major versions not only the UI changed, but also the API was moved to another place.

Would take the time to rewrite this, if no one is already working on it.

### PR DESCRIPTION
**Pull request by lumarel**
Fixes #2499 

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (planned Satellite 6.15)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* [ ] Foreman 3.4/Katello 4.6 (EL8 only)
* [ ] Foreman 3.3/Katello 4.5 on EL7 & EL8 (Satellite 6.12 on EL8 only; orcharhino 6.4/6.5 on EL8 only)
* We do not accept PRs for Foreman older than 3.3.

### CODE REVIEW

**Comment by ekohl** [comment]
While talking internally about this it was suggested that you can still configure Keycloak with `--http-relative-path` to add `/auth` back. In the short term this can be a note for the existing documentation.

**Comment by lumarel** [comment]
Okay will look into that and update accordingly 👍🏻

**Comment by lumarel** [comment]
@ekohl sorry that it took me that long to get back to this,
as we talked in the documentation meeting a few weeks back,
I split up the Wildfly and Quarkus stuff in it's own categories,
that of course generates a lot of duplicate text, if wanted I can split most of the duplicate stuff into snips 👍🏻

**Comment by asteflova** [comment]
> @ekohl sorry that it took me that long to get back to this, as we talked in the documentation meeting a few weeks back, I split up the Wildfly and Quarkus stuff in it's own categories, that of course generates a lot of duplicate text, if wanted I can split most of the duplicate stuff into snips 👍🏻

I'm currently reviewing and cleaning up the external authentication procedures. The ones for Keycloak are still on my to-do list, so if you want, you can leave any de-duplication for me.

> One thing, I was not sure about the guidance on when to set a punctuation at the end of a line, because I already saw both. My impression was that `*` does not have a punctuation, but all others do have it, is this correct?

If you're talking about lists (ordered or unordered), then the guidance we should follow in RH d/s is this:

* If a list item is a complete sentence, it should end with a period (.).
* If a list item is not a complete sentence, there is no period. Unless another item in that list is a complete sentence, in which case all list items should end with a period.

No need to give yourself a headache over this, I'll check the lists in your PR for this and add suggestions.

File: guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc
Comment (asteflova) [inline guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc:10]:
```suggestion
* A working installation of {Keycloak} server that uses HTTPS instead of HTTP.
* A {Keycloak} account with admin privileges.
* A realm for {Project} user accounts created in {Keycloak}.
* If the certificates or the CA are self-signed, ensure that they are added to the end-user certificate trust store.
* Users imported or added to {Keycloak}.
```
Periods at the end of each item because line 9 is a complete sentence.

File: guides/common/modules/con_prerequisites-for-configuring-project-with-wildfly-keycloak-authentication.adoc
Comment (asteflova) [inline guides/common/modules/con_prerequisites-for-configuring-project-with-wildfly-keycloak-authentication.adoc:10]:
```suggestion
* A working installation of {Keycloak} server that uses HTTPS instead of HTTP.
* A {Keycloak} account with admin privileges.
* A realm for {Project} user accounts created in {Keycloak}.
* If the certificates or the CA are self-signed, ensure that they are added to the end-user certificate trust store.
* Users imported or added to {Keycloak}.
```

File: guides/common/modules/proc_configuring-project-settings-for-quarkus-keycloak-authentication-using-the-cli.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-project-settings-for-quarkus-keycloak-authentication-using-the-cli.adoc:10]:
```suggestion
Note that you can navigate to the following URL within your realm to obtain values to configure {Project} settings: `\https://{Keycloak-short}.example.com/auth/realms/{Project}_Realm/.well-known/openid-configuration`

.Prerequisites

* Ensure that the *Client authentication* setting in the {Project} client in the {Keycloak} web UI is set to OFF.
```

File: guides/common/modules/proc_configuring-project-settings-for-quarkus-keycloak-authentication-using-the-web-ui.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-project-settings-for-quarkus-keycloak-authentication-using-the-web-ui.adoc:10]:
```suggestion
Note that you can navigate to the following URL within your realm to obtain values to configure {Project} settings: `\https://{Keycloak-short}.example.com/auth/realms/{Project}_Realm/.well-known/openid-configuration`

.Prerequisites

* Ensure that the *Client authentication* setting in the {Project} client in the {Keycloak} web UI is set to ON.
```

File: guides/common/modules/proc_configuring-the-project-client-in-quarkus-keycloak.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-the-project-client-in-quarkus-keycloak.adoc:11]:
```suggestion
* If you want your users to authenticate to {Project} using the {ProjectWebUI}, set the *Client authentication* setting to ON.
* If you want your users to authenticate to {Project} using the CLI, set the *Client authentication* setting to OFF.
```

File: guides/common/modules/proc_configuring-the-project-client-in-quarkus-keycloak.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-the-project-client-in-quarkus-keycloak.adoc:14]:
```suggestion
* If you want your users to authenticate to {Project} using the {ProjectWebUI}, in the blank field below the existing URI, enter a URI in the form `\https://{foreman-example-com}/users/extlogin`.
Note that you must add the string `/users/extlogin` after the {Project} FQDN.
```

File: guides/common/modules/proc_configuring-the-project-client-in-quarkus-keycloak.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-the-project-client-in-quarkus-keycloak.adoc:24]:
```suggestion
* If you want your users to authenticate to {Project} using the CLI, in the blank field below the existing URI, enter *urn:ietf:wg:oauth:2.0:oob*.
```

File: guides/common/modules/proc_configuring-the-project-client-in-wildfly-keycloak.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-the-project-client-in-wildfly-keycloak.adoc:11]:
```suggestion
* If you want your users to authenticate to {Project} using the {ProjectWebUI}, from the *Access Type* list, select *confidential*.
* If you want your users to authenticate to {Project} using the CLI, from the *Access Type* list, select *public*.
```

File: guides/common/modules/proc_configuring-the-project-client-in-wildfly-keycloak.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-the-project-client-in-wildfly-keycloak.adoc:14]:
```suggestion
* If you want your users to authenticate to {Project} using the {ProjectWebUI}, in the blank field below the existing URI, enter a URI in the form `\https://{foreman-example-com}/users/extlogin`. Note that you must add the string `/users/extlogin` after the {Project} FQDN.
```

File: guides/common/modules/proc_configuring-the-project-client-in-wildfly-keycloak.adoc
Comment (asteflova) [inline guides/common/modules/proc_configuring-the-project-client-in-wildfly-keycloak.adoc:24]:
```suggestion
* If you want your users to authenticate to {Project} using the CLI, in the blank field below the existing URI, enter *urn:ietf:wg:oauth:2.0:oob*.
```

**Review by asteflova — COMMENTED** [review]
A quick pass to check periods at the ends of list items.

**Comment by lumarel** [comment]
Hi @asteflova, thank you for taking the time for the review 🙂 

> I'm currently reviewing and cleaning up the external authentication procedures. The ones for Keycloak are still on my to-do list, so if you want, you can leave any de-duplication for me.

I saw your comment in an earlier PR about that, only just now remembered, missing to do a response there.
Okay great, there is for sure a lot of room for de-duplication here, as it's sometimes just 1 or 2 lines of the whole proc/con that needs to be different between Wildfly and Quarkus Keycloaks ^^

> If you're talking about lists (ordered or unordered), then the guidance we should follow in RH d/s is this:
>
> -   If a list item is a complete sentence, it should end with a period (.).
> -   If a list item is not a complete sentence, there is no period. Unless another item in that list is a complete sentence, in which case all list items should end with a period.
>
>No need to give yourself a headache over this, I'll check the lists in your PR for this and add suggestions.

Ahaha then I really just did an dis-improvement for some parts,
I remember to see a guideline page with all the different "how-to format" stuff, but I forgot where it was.
So I patched all the mentioned cases and re-did the whole commit, to remove these remove+add-again cases 👍🏻

**Comment by asteflova** [comment]
> Ahaha then I really just did an dis-improvement for some parts, I remember to see a guideline page with all the different "how-to format" stuff, but I forgot where it was. So I patched all the mentioned cases and re-did the whole commit, to remove these remove+add-again cases 👍🏻

No worries. At RH, we're actually following advice from The IBM Style guide on this, which is unfortunately a closed source and presumably simply unavailable to you. That's why I wouldn't ask from any member of the community to keep these rules in mind, and I'm totally fine applying them myself.

**Comment by asteflova** [comment]
> > I'm currently reviewing and cleaning up the external authentication procedures. The ones for Keycloak are still on my to-do list, so if you want, you can leave any de-duplication for me.
> 
> I saw your comment in an earlier PR about that, only just now remembered, missing to do a response there. Okay great, there is for sure a lot of room for de-duplication here, as it's sometimes just 1 or 2 lines of the whole proc/con that needs to be different between Wildfly and Quarkus Keycloaks ^^

Ok! We've done that before on this project -- various contributors merging their updates and a writer then submitting a follow-up PR to make a few more editing changes. So if you're comfortable with that, I can ack your PR from the _style point of view_, considering that I volunteer to review it more thoroughly myself within a few weeks.

What else is needed to wrap your PR up? We typically need an ack from the _technical point of view_ too.

**Review by asteflova — APPROVED** [review]

File: guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc
Comment (ekohl) [inline guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc:14]: We can use the `RHDocsBaseURL` attribute, but it appears this isn't done consistently throughout our codebase so not blocking IMHO.
```suggestion
For more information, see {RHDocsBaseURL}red_hat_build_of_keycloak/24.0/html/server_administration_guide/user-storage-federation[User Storage Federation] in the _Red{nbsp}Hat build of {Keycloak} Administration Guide_.
```

File: guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc
Comment (ekohl) [inline guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc:21]: Feels to me like this should be two sentences.
```suggestion
* This documentation relies on a {Keycloak} system setup with `--http-relative-path=/auth` configured. If you want to use a different context path, you will need to make manual adjustments or need to manually setup the httpd configuration.
```

**Review by ekohl — APPROVED** [review]
From a quick review I think this is correct, but it's a bit hard to see because of the duplication. The `--http-relative-path=/auth` step feels a bit hidden, but it's there.

**Comment by lumarel** [comment]
> From a quick review I think this is correct, but it's a bit hard to see because of the duplication. The `--http-relative-path=/auth` step feels a bit hidden, but it's there.

Hm I could make that bullet point in the requirements a WARNING box, but my thinking there was that it's as important as the certificate bullet points. Might this be better?

File: guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc
Comment (lumarel) [reply guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc]: Changed it for all URIs, might be worth a PR to change it for all at once for the rest 🙂

File: guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc
Comment (asteflova) [inline guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc:21]: If you want to help make sure this bullet point doesn't get overlooked, consider moving it to the user story introduction: assembly_configuring-quarkus-based....adoc. It doesn't seem like a prerequisite anyway -- a prerequisite is more of an an action to be taken _before_ you perform the configuration procedure; this, on the other hand, is more of a pre-existing condition, so better suited for the introduction. IMHO :)

Also, a few further suggestions regarding the wording:

```suggestion
* Configuring Quarkus-based {Keycloak} authentication requires the `--http-relative-path=/auth` context path to be configured in your {Keycloak} system setup.
If you want to use a different context path, make manual adjustments or set up the `httpd` service manually.
```

To explain why I'm recommending this:

* "This documentation" -- This can be confusing because "this documentation" can also actually apply to the whole guide, or everything on docs.theforeman.org... It's better to be specific about what part of documentation you're referring to: "Configuring Quarkus-based authentication"
* "you will need to" -- We tend to avoid future tense because simple present usually works just fine; also, imperative voice for actions to be take by the user is also shorter and easier to understand.
* "setup httpd configuration" -- 'setting up configuration' feels like using the same word twice

Comment (lumarel) [reply guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc:21]: > If you want to help make sure this bullet point doesn't get overlooked, consider moving it to the user story introduction: assembly_configuring-quarkus-based....adoc. It doesn't seem like a prerequisite anyway -- a prerequisite is more of an an action to be taken before you perform the configuration procedure; this, on the other hand, is more of a pre-existing condition, so better suited for the introduction. IMHO :)

Okay this would mean not directly into the assembly, but in the `con_configuring-quarkus-keycloak-authentication-for-project.adoc` because it applies to all 3 paths
The only thing to think about here then is, should we also directly move the certificate message up, because it's really the same case, if you don't have a Keycloak running on HTTPS it won't work.
Another option I'm thinking of is to make it the 2nd bullet point in the requirements, right after the HTTPS one?

> "setup httpd configuration" -- 'setting up configuration' feels like using the same word twice

The 1st 2, yep I will correct these, really might be the better way to write it :)
The 3rd, when thinking more about that, might even be better to write it even more exactly, because if the file-name of the config doesn't apply to the scheme of `${apache::confd_dir}/${keycloak_app_name}_oidc_keycloak_${keycloak_realm}.conf`, foreman-installer will just remove it on the next run.
Or is it best to keep it vague?

Comment (asteflova) [reply guides/common/modules/con_prerequisites-for-configuring-project-with-quarkus-keycloak-authentication.adoc:21]: > Okay this would mean not directly into the assembly, but in the `con_configuring-quarkus-keycloak-authentication-for-project.adoc` because it applies to all 3 paths The only thing to think about here then is, should we also directly move the certificate message up, because it's really the same case, if you don't have a Keycloak running on HTTPS it won't work. Another option I'm thinking of is to make it the 2nd bullet point in the requirements, right after the HTTPS one?

Ok, after staring at the list for a few more minutes, making it the 2nd bullet point does seem to make more sense :) A good idea.

> 
> > "setup httpd configuration" -- 'setting up configuration' feels like using the same word twice
> 
> The 1st 2, yep I will correct these, really might be the better way to write it :) The 3rd, when thinking more about that, might even be better to write it even more exactly, because if the file-name of the config doesn't apply to the scheme of `${apache::confd_dir}/${keycloak_app_name}_oidc_keycloak_${keycloak_realm}.conf`, foreman-installer will just remove it on the next run. Or is it best to keep it vague?

Agreed, sounds important enough to be mentioned.

**Comment by lumarel** [comment]
> > > From a quick review I think this is correct, but it's a bit hard to see because of the duplication. The `--http-relative-path=/auth` step feels a bit hidden, but it's there.
> > 
> > 
> > Hm I could make that bullet point in the requirements a WARNING box, but my thinking there was that it's as important as the certificate bullet points. Might this be better?
> 
> I took a closer look at this part and came up with an alternative suggestion. Feel free to accept it, decline, or use it to start a conversation to come up with a better solution :)

I tried to think it all through now, including the questions above, so this would be my proposal how I would write these few prerequisite lines, while trying to keep everything we talked about in mind 🙂

**Comment by asteflova** [comment]
Works for me @lumarel! Is there anything else you'd like to change in this PR before we merge it?

(BTW, as I mentioned, I plan to follow up on your PR with one of my own, as part of my ongoing efforts to consolidate all docs for external authentication, in collaboration with the RH Keycloak team. I intend to ask you for a review if you're still around in June/July.)

**Comment by lumarel** [comment]
@asteflova Think I caught everything that bugged me now, thanks for asking 🙂 

> (BTW, as I mentioned, I plan to follow up on your PR with one of my own, as part of my ongoing efforts to consolidate all docs for external authentication, in collaboration with the RH Keycloak team. I intend to ask you for a review if you're still around in June/July.)

Okay yeah sure, I will make sure to check my notifications ^^

**Comment by asteflova** [comment]
I'm cherry-picking this to 3.11:

7e2038628b..72013f0895  3.11 -> 3.11

 because my previous work on external authentication scenarios (beginning with https://github.com/theforeman/foreman-documentation/pull/2938) landed there too.

**Pull request merged**
